sqlglot/dialects/clickhouse.py

```diff
@@ -161,6 +163,14 @@ def _parse_table(
             joins: bool = False,
             alias_tokens: t.Optional[t.Collection[TokenType]] = None,
         ) -> t.Optional[exp.Expression]:
+            if (
+                self._index >= 2
+                and self._tokens[self._index - 2].token_type == TokenType.ARRAY
+                and self._prev.token_type == TokenType.JOIN
+                and self._curr.token_type == TokenType.L_BRACKET
+            ):  # e.g. ARRAY JOIN [1, 2, 3]
+                return self._parse_expression()
```

Comment:
I'm not sure if this representation fully captures the intent. Maybe we should instead:

1. Use a more specific method like `_parse_bracket`
2. Wrap the returned `Bracket` in a `Table`, so that we know it's used as a source
3. Try to `_parse_table_alias` and, if successful, attach the alias to the `Table` expression.